internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadWriteSchemaRepository.java

```diff
@@ -486,7 +487,9 @@ public Pair<Integer, Integer> getDerivedSchemaId(String storeName, String derive
         .stream()
         .flatMap(List::stream)
         .collect(Collectors.toList())) {
-      if (derivedSchemaEntry.getSchema().equals(derivedSchema)) {
+      String repoSchemaStr = AvroCompatibilityHelper.toParsingForm(derivedSchemaEntry.getSchema());
```

Comment:
This seems like an expensive call... can we add a field and getter for canonicalized strings into the `SchemaEntry` itself?